src/HttpClientRequestAdapter.cs

```diff
@@ -293,7 +293,19 @@ public ISerializationWriterFactory SerializationWriterFactory
                         {
                             result = rootNode.GetDateValue();
                         }
-                        else throw new InvalidOperationException("error handling the response, unexpected type");
+                        else
+                        {
+                            var underlyingType = Nullable.GetUnderlyingType(modelType);
+                            if(underlyingType != null && underlyingType.IsEnum)
+                            {
+                                var rawValue = rootNode.GetStringValue();
+                                result = Enum.Parse(underlyingType, rawValue!, true);
+                            }
+                            else
+                            {
+                                throw new InvalidOperationException("error handling the response, unexpected type");
+                            }
+                        }

```

Comment:
```suggestion
                        else if (!(
                            Nullable.GetUnderlyingType(modelType) is { IsEnum: true } underlyingType &&
                            rootNode.GetStringValue() is { Length :> 0 } rawValue &&
                            Enum.TryParse(underlyingType, rawValue, true, out result)
                        ))
                        {
                            throw new InvalidOperationException("error handling the response, unexpected type");
                        }
```

This should be possible, and it adds a bunch of additional checks